webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala

```diff
@@ -336,6 +336,7 @@ class ComplianceAPIService(
                     ByRuleNodeCompliance(
                       nodeId
```

Comment:
All the part in the yield is getting quite complicated (and where the actual business logic lies). 

Can you create a pure function that takes in argument all the IO parts obtained in the for loop and with the logic, so that it can be tested in isolation ?

Actually: I fear that just integration tests on these API will be hard, but perhaps my fear is misplaced and it's better to start with test at that level, even if harder to get right, they will be more meaningfull (and we need them in all cases)